src/H5TSint.c

```diff
@@ -135,26 +135,32 @@ H5TS__init(void)
  *
  * Purpose:  Terminate this interface.
  *
- * Note:     This function is currently registered via atexit() and is called
- *              AFTER H5_term_library().
- *
  * Return:    void
  *
  *-------------------------------------------------------------------------
  */
-void
+int
 H5TS_term_package(void)
 {
     FUNC_ENTER_NOAPI_NOINIT_NOERR
 
+    /* Clean up per-thread library info */
+    H5TS__tinfo_term();
+
+    /* Tell subsequent library opens that thread information is not initialized */
+    H5TS_first_init_s = H5TS_ONCE_INITIALIZER;
```

Comment:
This looks like a good idea, but I don't know if the standard(s) supports it, so I don't think it'll work.   That's why I put the H5TS_term_package as an atexit() callback, instead of when the library is closed.